jgrapht-ext/src/main/java/org/jgrapht/ext/GraphMLImporter.java

```diff
@@ -0,0 +1,704 @@
+                        handleSpecialEdgeWeights = true;
+                        String defaultValue = k.defaultValue;
+                        try {
+                            if (defaultValue != null) {
+                                defaultSpecialEdgeWeight = Double
+                                    .parseDouble(defaultValue);
+                            }
+                        } catch (NumberFormatException e) {
+                            // ignore
+                        }
+                        // first key only which maps to special edge "weight"
+                        break;
+                    }
+                }
+
+            }
+
+            // create edges
+            for (NodeOrEdge p : edges) {
+                V from = graphNodes.get(p.id1);
+                if (from == null) {
+                    throw new ImportException(
+                        "Source vertex " + p.id1 + " not found");
+                }
+                V to = graphNodes.get(p.id2);
+                if (to == null) {
+                    throw new ImportException(
+                        "Target vertex " + p.id2 + " not found");
+                }
+
+                // create attributes
+                Map<String, String> collectedAttributes = p.attributes;
+                Map<String, String> finalAttributes = new HashMap<String, String>();
+
+                for (Key validKey : edgeValidKeys.values()) {
+                    String validId = validKey.id;
+                    if (collectedAttributes.containsKey(validId)) {
+                        finalAttributes.put(
+                            validKey.attributeName,
+                            collectedAttributes.get(validId));
+                    } else {
+                        if (validKey.defaultValue != null) {
+                            finalAttributes.put(
+                                validKey.attributeName,
+                                validKey.defaultValue);
+                        }
+                    }
+                }
+
+                E e = edgeProvider.buildEdge(
+                    from,
+                    to,
+                    "e_" + from + "_" + to,
+                    finalAttributes);
+                graph.addEdge(from, to, e);
+
+                // special handling for weighted graphs
+                if (handleSpecialEdgeWeights) {
+                    if (finalAttributes.containsKey(edgeWeightAttributeName)) {
+                        try {
+                            ((WeightedGraph<V, E>) graph).setEdgeWeight(
+                                e,
+                                Double.parseDouble(
+                                    finalAttributes
+                                        .get(edgeWeightAttributeName)));
+                        } catch (NumberFormatException nfe) {
+                            ((WeightedGraph<V, E>) graph)
+                                .setEdgeWeight(e, defaultSpecialEdgeWeight);
```

Comment:
Should the reader print a warning wrt the  NumberFormatException, or escalate the exception?